src/vue/common/CollectionRenderer.vue

```diff
@@ -0,0 +1,88 @@
+<template>
+  <div class="pagination">
+    <button
+      class="pagination__more-button"
+      @click="loadData(false)">
+      {{ 'common.button-label-more' | globalize }}
+    </button>
+  </div>
+</template>
+
+<script>
+/**
+ * This component is responsible for downloading data in
+ * which there is a pagination. Component return loaded data with action.
```

Comment:
After loading <Component name> emits the event (describe what event) with the loaded data and decide whether or not it still show itself.

<ComponentName> doesn't modify data at all. You are responsible for doing that if needed. If empty response came, the empty array will be emitted